cpp/include/raft/spatial/knn/detail/ivf_flat_search.cuh

```diff
@@ -886,6 +886,7 @@ struct euclidean_dist<Veclen, uint8_t, uint32_t> {
       const auto diff = __vabsdiffu4(x, y);
       acc             = dp4a(diff, diff, acc);
     } else {
+      // Note: diff can underflow but the multiplication is still correct in that case.
```

Comment:
Now that you said it out loud, it feels a little uncomfortable. Maybe use `x > y ? x - y : y - x` or `__usad ( unsigned int  x, unsigned int  y, unsigned int  z)`?

Reply:
`x > y ? x - y : y - x` is less efficient than casting to signed.

I could go with `__usad`, it switches one instruction for another: `IADD3` to `VABSDIFF.U32`.